src/node_perf.cc

```diff
@@ -89,9 +89,9 @@ void PerformanceEntry::Notify(Environment* env,
   if (type != NODE_PERFORMANCE_ENTRY_TYPE_INVALID &&
       observers[type]) {
     node::MakeCallback(env->isolate(),
-                       env->process_object(),
+                       object.As<Object>(),
                        env->performance_entry_callback(),
-                       1, &object);
+                       1, &object, {0, 0});
```

Comment:
Question: what would the the correct context here? To me it seems that an `AsyncResource` or `async_context` source be created when the observer is constructed, and that should be used on the callback. Do you agree?

Aside: Do our compilers support `async_context{0,0}` syntax yet?

Reply:
Still thinking about that. `PerformanceObserver` currently is not an `AsyncResource` and only actually exists on the JS side. Making it an `AsyncResource` would be a minor divergence from the spec but is certainly not out of the question.

Reply:
And I don't know about the compilers question.

Reply:
If you don't want to inherit from `AsyncResource`, you can attache an `async_context` to the `PerformanceObserver` instances for the same effect.

I'm okay if you want to do this in a follow-on, as the issue existed before this PR, and is orthogonal.

Reply:
Added example to https://github.com/nodejs/node/pull/18810 in case it is useful. If the CI passes there, then that would answer the compiler question :).